File: 0492-construct-the-rectangle/0492-construct-the-rectangle.py
class Solution:
    def constructRectangle(self, area: int) -> List[int]:
        arr = []
        for i in range(1, int(area**0.5) + 1):  
            if area % i == 0:
                arr.append((i, area//i, abs(i - area//i)))
        
        
        arr.sort(key=lambda x: x[2])
        return [arr[0][1], arr[0][0]]  

File: 0492-construct-the-rectangle/0492-construct-the-rectangle.dart
class Solution {
  List<int> constructRectangle(int area) {
    List<List<int>> matrix = [];
    for(int i = 1 ; i < sqrt(area).toInt() + 1; i++){
        if(area % i == 0){
            matrix.add([i,area~/i,(i - area~/i).abs()]);
        }
    }
    matrix.sort((a, b) => a[2].compareTo(b[2]));
    return [matrix[0][1], matrix[0][0]];

  }
}